rust/gitxetcore/src/command/summary.rs

```diff
@@ -103,8 +105,20 @@ async fn print_summary_from_db(
     match summary_type {
         SummaryType::Libmagic => print_stored_summary_impl(&summary.libmagic),
         SummaryType::Csv => print_stored_summary_impl(&summary.csv),
-        SummaryType::Twb => print_stored_summary_impl(&summary.twb),
-        SummaryType::Tds => print_stored_summary_impl(&summary.tds),
+        SummaryType::Twb => {
+            if let Some(sum) = summary.additional_summaries.as_ref() {
```

Comment:
nit:  Could do `print_stored_summary_impl(summary.additional_summaries.and_then(|s| s.twb))`

Reply:
Not really, `summary` is a reference, so we need to have a reference `.as_ref()` to call the `and_then` (since `and_then` takes ownership of the option). If we grab the references for `additional_summaries` and `s.twb`, then we are stuck with an `Option<&TwbSummary>` instead of an `&Option<TwbSummary>` (&TwbSummary doesn't implement Default, only `TwbSummary`).